### DISCUSSION TITLE
Linkerd with CNI Plugin coexisting with another CNI?

### DISCUSSION THREAD

**Discussion by FrederikNJS** [linkerd/linkerd2#6992] [Q&A] [answered]
Hi Linkerd,

I was wanting to try out Linkerd in our clusters. I would like to install using the CNI options, to avoid needing the init containers. It's a bit unclear from the documentation whether Linkerd then takes over the entire CNI, or whether it just complements the already existing CNI. I currently use Calico, and was worried that Linkerd would replace Calico. And instead hoping that Linkerd CNI would coexist nicely with Calico, simply augmenting pods that make use of Linkerd. How does the Linkerd CNI plugin interact with other CNIs?

**Comment by wmorgan** [comment, marked as answer]
Hi @FrederikNS. I've tried to clarify this in the docs. The Linkerd CNI plugin will not replace Calico, and will integrate with it using CNI chaining. You should use both together. Let us know if that doesn't work!
